common/src/main/java/io/pravega/common/util/TypedProperties.java

```diff
@@ -125,27 +125,35 @@ public boolean getBoolean(Property<Boolean> property) throws ConfigurationExcept
     }
 
     private <T> T tryGet(Property<T> property, Function<String, T> converter) {
-        String fullKeyName = this.keyPrefix + property.getName();
+        String propNewName = this.keyPrefix + property.getName();
+        String propOldName = this.keyPrefix + property.getLegacyName();
+        String propValue = null;
 
-        // Get value from config.
-        String value = this.properties.getProperty(fullKeyName, null);
+        if (property.hasLegacyName()) {
+            // Value of property with old name
+            propValue = this.properties.getProperty(propOldName, null);
+        }
 
-        if (value == null && property.hasLegacyName()) {
-            value = this.properties.getProperty(this.keyPrefix + property.getLegacyName(), null);
+        if (propValue == null) {
+            // Value of property with new name
+            propValue = this.properties.getProperty(propNewName, null);
```

Comment:
Same here, please call out it's new property name being used.

Reply:
Consider adding code to handle the scenario that both legacy and current properties are being defined.
This is the most confusing scenario, so please log it in details, e.g. "both legacy and current properties defined as blah blah, and legacy properties is used."

Reply:
I disagree. See below.

Just log the most important thing, which is when you are still using something that will not be supported in the future.

Reply:
I agree with @andreipaduroiu here.